cmd/build_test.go

```diff
@@ -2002,3 +2002,118 @@ foo contains __local1__1 if {
 		})
 	}
 }
+
+// TestBuildWithFollowSymlinks tests that the build command follows symlinks when building a bundle.
+// This test uses a local tmp filesystem to create a directory with a symlink to a file in it's root
+// and a local file in the bundle directory, and verifies that the built bundle contains both the symlink
+// and the regular file.
+// There's probably some common utilities that could be extracted at some point but for now this code is
+// local to the test until we need to reuse it elsewhere.
+func TestBuildWithFollowSymlinks(t *testing.T) {
+	rootDir, err := os.MkdirTemp("", "build-follow-symlinks")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := os.RemoveAll(rootDir); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	bundleDir := path.Join(rootDir, "bundle")
+	err = os.Mkdir(bundleDir, 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create a regular file in our temp bundle directory
+	err = os.WriteFile(filepath.Join(bundleDir, "foo.rego"), []byte("package foo\none = 1"), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create a regular file in the root directory of our tmp directory that we will symlink into the bundle directory later
+	err = os.WriteFile(filepath.Join(rootDir, "bar.rego"), []byte("package foo\ntwo = 2"), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create a symlink in the bundle directory to the file in the root directory
+	err = os.Symlink(filepath.Join(rootDir, "bar.rego"), filepath.Join(bundleDir, "bar.rego"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	params := newBuildParams()
+	params.outputFile = path.Join(rootDir, "test.tar.gz")
+	params.bundleMode = true
```

Comment:
Can we drop This? Reading in bundle-mode doesn't seem like a requirement.

Reply:
👍

Reply:
Hm, I think I might have lead you astray with this coment 🤔 .
If I add `params.bundleMode = true` here, the test starts to fail.
But if I also add `params.followSymlinks = true`, it succeeds again, clearly showing that your code is being exercised. 

So we need to add 

```
params.bundleMode = true
params.followSymlinks = true
```